### DISCUSSION TITLE
[Meta] Change licence to AGPL or other copyleft licence

### DISCUSSION THREAD

**Discussion by filipesmedeiros** [cheeaun/phanpy#346] [General]
Hey! Really love the project. I think you know me by now ahah

Just wanted to explore the option of changing the licence of the project to [AGPL](https://www.gnu.org/licenses/agpl-3.0.en.html) or another [copyleft](https://en.wikipedia.org/wiki/Copyleft) licence. Phanpy has no official company behind it (as far as I know), and protecting it from being stolen by profit-seeking actors seems valuable.

Do you see any advantage in keeping it MIT?

Thanks for your time!

**Comment by cheeaun** [comment]
Didn't really think much about advantages of MIT or any licences. Yes, there's no company. Though I can't imagine how it could be "stolen" too, despite that I do encourage folks to "steal" any/all the ideas and experiments implemented on Phanpy 😬

**Comment by filipesmedeiros** [comment]
@cheeaun I guess it's just the difference between sharing and being taken advantage of. Since MIT allows closed source versions, someone could just use Phanpy's code to create a closed source, for profit company, pour a bunch of money into it and take over your hard work. I'm not saying it's _gonna_ happen. But you can't be too sure ahah

Plus, AGPL incentivises building for the common good, not for selfish motives only (of course selfish and common most of the times align!)

All of this said, the project is yours and you decide what you feel is the purpose of it :)

**Comment by cheeaun** [comment]
@filipesmedeiros totally understand your points. I'm also aware of news from other open-source projects changing to AGPL too.

To be honest, as of now, I don't have any strong concerns on the licensing.

Some experiments and UI ideas on Phanpy are very user-focused and not engagement-driven. They are also not *new* ideas e.g. other social media apps use carousels for inserting out-of-context content (ads, cross-product materials, random post from strangers) in the timeline, but Phanpy use carousels (Boost Carousel) to **reduce** visual clutter and distractions instead — complete **opposite** of their intentions/motives/goals/KPI. I would personally *want* them to copy these 😂

**Reply by filipesmedeiros** [reply]
Makes sense! I have no problem with this :)
